giskard/models/base/model.py

```diff
@@ -598,3 +619,133 @@ def __str__(self) -> str:
+            The dictionary with Tools' names and related instances.
+        """
+        return {
+            PredictTool.default_name: PredictTool(model=self, dataset=dataset),
+            MetricTool.default_name: MetricTool(model=self, dataset=dataset),
+            SHAPExplanationTool.default_name: SHAPExplanationTool(model=self, dataset=dataset),
+            IssuesScannerTool.default_name: IssuesScannerTool(scan_result=scan_report),
+        }
+
+    @staticmethod
+    def _gather_context(message_list: list) -> str:
+        """Gather context into a single string.
+
+        Given the list of OpenAI's messages, extracts and joins their contents into a single string.
+
+        Parameters
+        ----------
+        message_list : list
+            The list of OpenAI's messages.
+
+        Returns
+        -------
+        str
+            The string with the joined messages' contents.
+        """
+        context = list()
+
+        for msg in message_list:
+            if isinstance(msg, dict) and "content" in msg:
+                context.append(json.dumps(msg))
+
+        return "\n".join(context)
+
+    def talk(
+        self, question: str, context: str = "", dataset: Dataset = None, scan_report: "ScanReport" = None
+    ) -> TalkResult:
+        """Perform the 'talk' to the model.
+
+        Given `question`, allows to ask the model about prediction result, explanation, model performance, issues, etc.
+
+        Parameters
+        ----------
+        question : str
+            User input query.
+        context : str
+            Context of the previous 'talk' results. Necessary to keep context between sequential 'talk' calls.
+        dataset : Dataset
+            Giskard Dataset to be analysed by the 'talk'.
+        scan_report : ScanReport
+            Giskard Scan Report to be analysed by the 'talk'.
+        """
+        set_llm_model(LLM_MODEL)
+        client = get_default_client()
+
+        available_tools = self._get_available_tools(dataset, scan_report)
+
+        system_prompt = MODEL_INSTRUCTION.format(
+            tools_description="\n".join([tool.description for tool in list(available_tools.values())]),
+            model_name=self.meta.name,
+            model_description=self.meta.description,
+            feature_names=self.meta.feature_names,
+            context=context,
+        )
+
+        messages = [{"role": "system", "content": system_prompt}, {"role": "user", "content": question}]
+
+        response = client.complete(
+            messages=messages,
+            tools=[tool.specification for tool in list(available_tools.values())],
+            tool_choice="auto",
+            temperature=0.1,
+            max_tokens=MAX_COMPLETION_TOKENS,
+        )
+
+        if content := response.content:
+            messages.append({"role": "assistant", "content": content})
+
+        if tool_calls := response.tool_calls:
+            messages.append(BaseOpenAIClient._serialize_message(response))
```

Comment:
I think you should be using client._serialize_message here

Reply:
ok